dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4380,6 +4380,417 @@ class FunctionRepeat : public IFunction
     }
 };
 
+class FunctionInstrUTF8 : public IFunction
+{
+public:
+    static constexpr auto name = "instrUTF8";
+
+    FunctionInstrUTF8() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionInstrUTF8>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 2; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        auto first_argument = removeNullable(arguments[0]);
+        if (!first_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        auto second_argument = removeNullable(arguments[1]);
+        if (!second_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        auto return_type = std::make_shared<DataTypeInt64>();
+        return (arguments[0]->isNullable() || arguments[1]->isNullable()) ? makeNullable(return_type) : return_type;
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        auto c0_col_column = block.getByPosition(arguments[0]).column;
+        auto c1_col_column = block.getByPosition(arguments[1]).column;
+        size_t val_num = block.rows();
+
+        auto col_res = ColumnInt64::create();
+        col_res->reserve(val_num);
+        ColumnInt64::Container & res_vec = col_res->getData();
+        res_vec.resize(val_num);
+
+        const auto * str_const_string = checkAndGetColumnConst<ColumnString>(c0_col_column.get());
+        const auto * substr_const_string = checkAndGetColumnConst<ColumnString>(c1_col_column.get());
+        const auto * str_vector_string = checkAndGetColumn<ColumnString>(c0_col_column.get());
+        const auto * substr_vector_string = checkAndGetColumn<ColumnString>(c1_col_column.get());
```

Comment:
I think we can put these codes into if() clause. such as:

if ((const auto * str_const_string = checkAndGetColumnConst<ColumnString>(c0_col_column.get()))) {...}

so that we can reduce many useless `checkAndGetColumn`.